src/cargo/sources/git/utils.rs

```diff
@@ -1221,9 +1228,21 @@ fn fetch_with_cli(
         .env_remove("GIT_OBJECT_DIRECTORY")
         .env_remove("GIT_ALTERNATE_OBJECT_DIRECTORIES")
         .cwd(repo.path());
+
     gctx.shell()
         .verbose(|s| s.status("Running", &cmd.to_string()))?;
-    cmd.exec()?;
+
+    let revspec_id: String = refspecs
+        .iter()
+        .map(|s| s.split(':').next().unwrap_or("").to_string())
+        .collect();
+
+    cmd.exec().with_context(|| {
+        format!(
+            "Failed to fetch via Github fast path \nrevspec {:?} not found",
```

Comment:
BTW, this does not look related to the GitHub fast path. I'm not sure I understand why it is mentioning it here.